### DISCUSSION TITLE
on_ground判定

### DISCUSSION THREAD

**Discussion by Naoki-Hiraoka** [YutaKojio/hrpsys-base#24] [Q&A] [answered]
on_ground判定は、ロボットが受ける鉛直方向力の和が50N以上かどうかでなさるのだとずっと思っていたのですが、実際は、pref_act_force_zはエンドエフェクタのローカル座標系なので、ローカル座標系Z成分の値の和が50N以上かどうかで行っていました。
https://github.com/YutaKojio/hrpsys-base/blob/5aab7f9809b1ef76dd62bf323e522834d84097be/rtc/AutoBalancer/Stabilizer.cpp#L1309-L1311

環境との接触が全て面接触であればどちらでも良い? そもそもジャンプ中などはon_ground判定はどう扱っているのでしょう?

**Comment by YutaKojio** [comment]
> ロボットが受ける鉛直方向力の和が50N以上かどうかでなさるのだとずっと思っていたのですが

ちょうどこの前end-effector座標だとまずいなと思って，手元で鉛直方向にするようにしてシミュレーションで試しましたが，
歩行だと全然変わらないからまあいやとしてました．
でも直しておいて損はないので後でそうしておきます．

> そもそもジャンプ中などはon_ground判定はどう扱っているのでしょう?

ジャンプ中も同じくだけど，どうまずいだろう？

**Comment by Naoki-Hiraoka** [comment]
> ジャンプ中も同じくだけど，どうまずいだろう

MODE_AIRを、単にクレーンに吊るしたときにStablizerを止めるためのモードだと思っていたので、ジャンプ中にMODE_AIRに入って止まってしまったら困りそうだなと思ったのですが、ジャンプ中は接触力を利用した制御はできないので意図的に止めているということでしょうか。

**Comment by YutaKojio** [comment]
MODE_AIR判定になるまでの時間をジャンプに比べて十分長く設定しています。
https://github.com/YutaKojio/rtmros_tutorials/blob/ad4472615a82e0d68c25b9f3cc1b6dd59f36edbd/hrpsys_ros_bridge_tutorials/src/hrpsys_ros_bridge_tutorials/urata_hrpsys_config.py#L244

detection_count_to_airを長くすると、ST入ったままクレーンで釣ってMODE_AIR判定で自動で切るという運用をしたいときに困りますが、
そもそもJAXON系では僕のブランチmasterブランチ関係なく、stop-stしてから上げるというやり方しか近年行われていない気がします。

**Comment by YutaKojio** [comment, marked as answer]
> でも直しておいて損はないので後でそうしておきます．

https://github.com/YutaKojio/hrpsys-base/commit/a6f1a9f1217d43f91fe477b013d79058b85ff5d3
で変更しました．
ちなみに，prev_act_forceを3次元にしたのは地面についたのか段差の横に引っかかったのかなどを切り分けられるようにしたほうがいいかなと思い，すぐ試せるようにそうしました．
